packages/samples/headless-react/src/components/dependent-facet/complex-dependencies.fn.tsx

```diff
@@ -0,0 +1,81 @@
+import {useEffect, useState} from 'react';
+import {Facet as HeadlessFacet, FacetState} from '@coveo/headless';
+import {Facet} from '../facet/facet.fn';
+
+interface ComplexDependency {
+  [facetId: string]: {
+    facet: HeadlessFacet;
+    controls?: HeadlessFacet[];
+    dependsOn?: HeadlessFacet;
```

Comment:
Yes, you are right that just one of the two property is actually needed.

However, removing one of the two simplify the config in the sample page, but will make the code sample a bit harder to understand IMO.

To build the sample, we need to subscribe only to parent change, since they are the only one that need to be subscribed to to trigger the `useEffect` dependency that is needed to update the relationship. 

So if we had only `dependsOn`, then it becomes a bit more convoluted instead of directly looking at each individual facet and see if they "control" something.

On the flip side, it's simpler to define the "should a facet show" condition when you think about it from each individual facet perspective, and check if they depend on something to show, versus having each parent do the job.

Said differently, if we were to offer a prebuilt Atomic (react or web) component for this, then of course the thing we'd want to optimize is to make the config as simple as possible, and we would not require `controls` and `dependsOn`, just one of the two. 

But here I think the star of the show should be helping understand the "relationship".

What do you think ?

Reply:
I think it would be clearer if the sample just used `dependsOn`, especially if it is simpler to implement (i.e. it's Coveo's recommendation to use a `dependsOn` approach). Here's the current config from the `Page` component with annotations:

```
[this.sourceFacet.state.facetId]: {
  facet: this.sourceFacet,
  controls: [this.objectTypeFacet, this.fileTypeFacet], // it's not clear this facet is the parent because it might also be specified by the "controls" of another facet. I need to check the others to make sure.
},
[this.objectTypeFacet.state.facetId]: {
  facet: this.objectTypeFacet,
  dependsOn: this.sourceFacet,
},
[this.authorFacet.state.facetId]: {
  facet: this.authorFacet,
  dependsOn: this.sourceFacet,
  controls: [this.authorFacet], // is this intentional for a facet to control itself?
},
[this.fileTypeFacet.state.facetId]: {
  facet: this.fileTypeFacet,
  dependsOn: this.authorFacet,
},
```
vs. without `controls`

```
[this.sourceFacet.state.facetId]: {
  facet: this.sourceFacet,
},
[this.objectTypeFacet.state.facetId]: {
  facet: this.objectTypeFacet,
  dependsOn: this.sourceFacet,
},
[this.authorFacet.state.facetId]: {
  facet: this.authorFacet,
  dependsOn: this.sourceFacet,
},
[this.fileTypeFacet.state.facetId]: {
  facet: this.fileTypeFacet,
  dependsOn: this.authorFacet,
},
```

Alternatively, if we expect some clients to request an example of how to implement a `controls` approach, then we can create a dedicated sample show-casing it. I think show-casing only one way would reduce confusion though.

Reply:
I find having references to both a parent and a child confusing, and in this case it's similar kind of relationship. When I read this sample, and especially when I read `Complex`, I'm a bit confused about what's supposed to happen, unfortunately. Perhaps this sample could benefit from some comments, if we can't simplify it enough.